models/optimism/optimism/ovm_optimism_l2_token_factory.sql

```diff
@@ -0,0 +1,72 @@
+{{ config(
+        schema = 'ovm_optimism'
+        , alias='l2_token_factory'
+        , materialized = 'incremental'
+        , file_format = 'delta'
+        , incremental_strategy = 'merge'
+        , unique_key = ['l1_token', 'l2_token']
+        , post_hook='{{ expose_spells(\'["optimism"]\',
+                                  "project",
+                                  "ovm_optimism",
+                                  \'["msilb7"]\') }}'
+        ,depends_on=['tokens_optimism_erc20','tokens_erc20']
```

Comment:
Also when I compile the code locally, I get this error message:
```
RuntimeError: Found a cycle: model.spellbook.ovm_optimism_l2_token_factory --> model.spellbook.tokens_optimism_erc20 --> model.spellbook.tokens_erc20 --> snapshot.spellbook.tokens_erc20_snapshot
```

Reply:
yeah this is the main issue I'm having. Not sure the best way to handle for this in dbt, so figured we'd do it here vs me trying random stuff. 

My ideal flow:
1. Aave, Granary, L2 Token Factory, etc generated token lists read from tokens_erc20
2. These generated token list tables update
3. erc20_tokens (optimism version) then reads from these new generated token lists to create the final table.

Reply:
updated to remove the circular-ness for now.